components/button/button-icon.js

```diff
@@ -174,7 +176,7 @@ class ButtonIcon extends ThemeMixin(ButtonMixin(VisibleOnAncestorMixin(RtlMixin(
 				name="${ifDefined(this.name)}"
 				title="${ifDefined(this.text)}"
 				type="${this._getType()}">
-				<d2l-icon icon="${ifDefined(this.icon)}" class="d2l-button-icon"></d2l-icon>
+				<slot name="icon"><d2l-icon icon="${ifDefined(this.icon)}"></d2l-icon></slot>
```

Comment:
So I've removed the `d2l-button-icon` class. Based on a search this seemed safe but I can put it back if there are concerns.

Reply:
I'm good with that!

Thoughts about just conditionally rendering the `<d2l-icon>`? I'm not sure how the browser does this, but I'd hate for it to quickly render with no icon and then get replaced by the custom one.

Something like:

```javascript
const icon = this.icon ? html`<d2l-icon icon="${this.icon}"></d2l-icon>` : nothing;
return html`
  <button
    ...>
    <slot name="icon">${icon}</slot>
  </button>`;
```

Reply:
Ah true, makes sense